{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "tensorflow",
                    "vendor": "tensorflow",
                    "versions": [
                        {
                            "status": "affected",
                            "version": "< 1.15.4"
                        },
                        {
                            "status": "affected",
                            "version": ">= 2.0.0, < 2.0.3"
                        },
                        {
                            "status": "affected",
                            "version": ">= 2.1.0, < 2.1.2"
                        },
                        {
                            "status": "affected",
                            "version": ">= 2.2.0, < 2.2.1"
                        },
                        {
                            "status": "affected",
                            "version": ">= 2.3.0, < 2.3.1"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "In tensorflow-lite before versions 1.15.4, 2.0.3, 2.1.2, 2.2.1 and 2.3.1, when determining the common dimension size of two tensors, TFLite uses a `DCHECK` which is no-op outside of debug compilation modes. Since the function always returns the dimension of the first tensor, malicious attackers can craft cases where this is larger than that of the second tensor. In turn, this would result in reads/writes outside of bounds since the interpreter will wrongly assume that there is enough data in both tensors. The issue is patched in commit 8ee24e7949a203d234489f9da2c5bf45a7d5157d, and is released in TensorFlow versions 1.15.4, 2.0.3, 2.1.2, 2.2.1, or 2.3.1."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "HIGH",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "NONE",
                        "baseScore": 7.4,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "HIGH",
                        "integrityImpact": "HIGH",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:N",
                        "version": "3.1"
                    }
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-125",
                            "description": "{\"CWE-125\":\"Out-of-bounds Read\"}",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                },
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-787",
                            "description": "{\"CWE-787\":\"Out-of-bounds Write\"}",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2020-10-29T15:06:18",
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/tensorflow/tensorflow/releases/tag/v2.3.1"
                },
                {
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-mxjj-953w-2c2v"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/tensorflow/tensorflow/commit/8ee24e7949a203d234489f9da2c5bf45a7d5157d"
                },
                {
                    "name": "openSUSE-SU-2020:1766",
                    "tags": [
                        "vendor-advisory",
                        "x_refsource_SUSE"
                    ],
                    "url": "http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00065.html"
                }
            ],
            "source": {
                "advisory": "GHSA-mxjj-953w-2c2v",
                "discovery": "UNKNOWN"
            },
            "title": "Data corruption in tensorflow-lite",
            "x_legacyV4Record": {
                "CVE_data_meta": {
                    "ASSIGNER": "security-advisories@github.com",
                    "ID": "CVE-2020-15208",
                    "STATE": "PUBLIC",
                    "TITLE": "Data corruption in tensorflow-lite"
                },
                "affects": {
                    "vendor": {
                        "vendor_data": [
                            {
                                "product": {
                                    "product_data": [
                                        {
                                            "product_name": "tensorflow",
                                            "version": {
                                                "version_data": [
                                                    {
                                                        "version_value": "< 1.15.4"
                                                    },
                                                    {
                                                        "version_value": ">= 2.0.0, < 2.0.3"
                                                    },
                                                    {
                                                        "version_value": ">= 2.1.0, < 2.1.2"
                                                    },
                                                    {
                                                        "version_value": ">= 2.2.0, < 2.2.1"
                                                    },
                                                    {
                                                        "version_value": ">= 2.3.0, < 2.3.1"
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                },
                                "vendor_name": "tensorflow"
                            }
                        ]
                    }
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "eng",
                            "value": "In tensorflow-lite before versions 1.15.4, 2.0.3, 2.1.2, 2.2.1 and 2.3.1, when determining the common dimension size of two tensors, TFLite uses a `DCHECK` which is no-op outside of debug compilation modes. Since the function always returns the dimension of the first tensor, malicious attackers can craft cases where this is larger than that of the second tensor. In turn, this would result in reads/writes outside of bounds since the interpreter will wrongly assume that there is enough data in both tensors. The issue is patched in commit 8ee24e7949a203d234489f9da2c5bf45a7d5157d, and is released in TensorFlow versions 1.15.4, 2.0.3, 2.1.2, 2.2.1, or 2.3.1."
                        }
                    ]
                },
                "impact": {
                    "cvss": {
                        "attackComplexity": "HIGH",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "NONE",
                        "baseScore": 7.4,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "HIGH",
                        "integrityImpact": "HIGH",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:N",
                        "version": "3.1"
                    }
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "{\"CWE-125\":\"Out-of-bounds Read\"}"
                                }
                            ]
                        },
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "{\"CWE-787\":\"Out-of-bounds Write\"}"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/tensorflow/tensorflow/releases/tag/v2.3.1",
                            "refsource": "MISC",
                            "url": "https://github.com/tensorflow/tensorflow/releases/tag/v2.3.1"
                        },
                        {
                            "name": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-mxjj-953w-2c2v",
                            "refsource": "CONFIRM",
                            "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-mxjj-953w-2c2v"
                        },
                        {
                            "name": "https://github.com/tensorflow/tensorflow/commit/8ee24e7949a203d234489f9da2c5bf45a7d5157d",
                            "refsource": "MISC",
                            "url": "https://github.com/tensorflow/tensorflow/commit/8ee24e7949a203d234489f9da2c5bf45a7d5157d"
                        },
                        {
                            "name": "openSUSE-SU-2020:1766",
                            "refsource": "SUSE",
                            "url": "http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00065.html"
                        }
                    ]
                },
                "source": {
                    "advisory": "GHSA-mxjj-953w-2c2v",
                    "discovery": "UNKNOWN"
                }
            }
        }
    },
    "cveMetadata": {
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "assignerShortName": "GitHub_M",
        "cveId": "CVE-2020-15208",
        "datePublished": "2020-09-25T18:45:40",
        "dateReserved": "2020-06-25T00:00:00",
        "dateUpdated": "2020-10-29T15:06:18",
        "state": "PUBLISHED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}